portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,71 +25,57 @@
 
-    public void jsonInit() {
-        hardCodedJson.add("This is my first comment!");
-        hardCodedJson.add("This is my 2nd comment!");
-        hardCodedJson.add("This is my 3rd and last comment!");
-
-        // String json1 = "{";
-        // json1 += "\"name\": ";
-        // json1 += "\"" + "Murilo" + "\"";
-        // json1 += "}";
-
-        // String json2 = "{";
-        // json2 += "\"age\": ";
-        // json2 += "\"" + "20" + "\"";
-        // json2 += "}";
-
-        // String json3 = "{";
-        // json3 += "\"gender\": ";
-        // json3 += "\"" + "Male" + "\"";
-        // json3 += ", ";
-        // json3 += "\"height\": ";
-        // json3 += "\"" + "1.75" + "\"";
-        // json3 += "}";
-
-        // hardCodedJson.add(json1);
-        // hardCodedJson.add(json2);
-        // hardCodedJson.add(json3);
-    }
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String name = (String) entity.getProperty("name");
+        String text = (String) entity.getProperty("text");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        Comment comment = new Comment(name, text, timestamp);
+        comments.add(comment);
+        }
+
+        Gson gson = new Gson();
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // String json = convertArrayListToJson(hardCodedJson);
-        String json = convertToJsonUsingGson(comments);
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(gson.toJson(comments));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         // Get the input from the form.
+        long timestamp = System.currentTimeMillis();
         String name = getParameter(request, "name", "");
         String text = getParameter(request, "comment", "");
+        Comment comment = new Comment(name, text, timestamp);
 
         Entity commentEntity = new Entity("Comment");
         commentEntity.setProperty("name", name);
         commentEntity.setProperty("text", text);
+        commentEntity.setProperty("timestamp", timestamp);
 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         datastore.put(commentEntity);
         
-        Comment comment = new Comment(name, text);
-        comments.add(comment);
-
-        String json = convertToJsonUsingGson(comments);
+        String json = convertToJsonUsingGson(comment);
```

Comment:
In the `doGet` method you're using gson directly, here you still have a helper method. Can you make this consistent between the two?

(You can declare a field `private final Gson gson = new Gson();` in the class and access it from both places, that'd be my suggestion. This avoids creating a new Gson object on every request and makes it easy to configure it in e.g. the constructor and affect all JSON serialization.)